### PR TITLE
Feature/member dashboard

### PR DESCRIPTION
**Pull request by ad2969**
Screen Shot 2020-07-02 at 6 53 35 PM

Additional changes since PR approved:
- Ability to search for events
- Mobile view (with chips instead of the top and side panels)

Screen Shot 2020-08-01 at 4 25 41 AM

Screen Shot 2020-08-01 at 4 26 56 AM

### CODE REVIEW

File: src/components/EventUserTable.js [hunk 1]
```diff
@@ -66,7 +66,8 @@ export class EventUserTable extends Component {
         const heardFrom = {}
         response.data.forEach(user => {
           if (user.heardFromData) {
-            heardFrom[user.heardFrom] = heardFrom.hasOwnProperty(user.heardFrom) ? heardFrom[user.heardFrom] + 1 : 1
+            heardFrom[user.heardFrom] = Object.prototype.hasOwnProperty.call(heardFrom, user.heardFrom)
```
Comment (ianmah) [inline src/components/EventUserTable.js:69]: I don't think we need `hasOwnProperty`, does this work?
```suggestion
            heardFrom[user.heardFrom] = heardFrom[user.heardFrom] ? heardFrom[user.heardFrom] + 1 : 1
```

File: src/pages/member/UserEvents.js [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+  useEffect(() => {
+    if (!props.events) updateEvents()
+    if (user.id) updateRegisteredEvents(user.id)
+  }, [])
+
+  const handleFavoriteEvent = async (eventId, toggle) => {
+    const body = { eventID: eventId, isFavourite: toggle }
+    await fetchBackend(`/users/favEvent/${user.id}`, 'PATCH', body)
+    const newEventsId = toggle
```
Comment (ianmah) [inline src/pages/member/UserEvents.js:126]: `newEventIds`?

Comment (ianmah) [reply src/pages/member/UserEvents.js:126]: and for the other places as well?

File: src/pages/member/UserEvents.js [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+      ? [...user.favedEventsID, eventId]
+      : user.favedEventsID.filter((id) => id !== eventId)
+    props.setUser({
+      ...user,
+      favedEventsID: newEventsId
+    })
+  }
+
+  const handleTabChange = (event, newIndex) => {
```
Comment (ianmah) [inline src/pages/member/UserEvents.js:135]: nit: unused param?

File: src/pages/member/UserEvents.js [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+              <ListItemText><Search fontSize='small' />&nbsp;All</ListItemText>
+            </ListItem>
+          </List>
+        </div>
+        <div style={styles.tabs.container}>
+
+          <Tabs
+            value={tabIndex}
+            indicatorColor='primary'
+            textColor='primary' // TODO: Set the primary color theme
```
Comment (ianmah) [inline src/pages/member/UserEvents.js:224]: can we set the primary colour now?

File: src/pages/member/UserEvents.js [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+                  handleCardClick={redirectToEvent}
+                  handleFavorite={handleFavoriteEvent}
+                  cardStyle={{ width: '40%' }}
+                />
+              ))}
+            </div>
+          </EventPanel>
+          <EventPanel currentIndex={tabIndex} index={TAB_STATES.PAST}>
+            <div style={styles.rows}>
+              {PastEventCards.map((event) => (
```
Comment (ianmah) [inline src/pages/member/UserEvents.js:250]: could we check the index here to decide which array to use? so we don't have to have duplicate as much code?

File: src/pages/member/UserEvents.js (hunk 3 above)
Comment (ad2969) [reply src/pages/member/UserEvents.js:135]: The material.ui function automatically passes the first `event` argument which we don't really need.. I'm not sure if there is a way to make this look any cleaner, is there?

File: src/pages/member/UserEvents.js (hunk 4 above)
Comment (ad2969) [reply src/pages/member/UserEvents.js:224]: Oh ya Derek's PR helped solve this [here](https://github.com/ubc-biztech/bt-web/blob/7cbc49779a1a27e1b168b73051a938a753cf9404/src/components/ThemeProvider.js#L62-L64) 👍

File: src/pages/member/UserEvents.js (hunk 5 above)
Comment (ad2969) [reply src/pages/member/UserEvents.js:250]: Hm.. Does turning it into its own function help?

**Comment by ad2969** [comment]
Included changes to allow public access for event register page (+ loading buffer and a "NotFound" if cannot access")

The changes requires `Authorizer` to be removed from the backend GET call (altho this may call for privacy issues? if we don't want people accessing data on our events)

The supposed backend PR is [here](https://github.com/ubc-biztech/serverless-biztechapp/pull/97)

File: src/pages/member/UserEvents.js (hunk 5 above)
Comment (ianmah) [reply src/pages/member/UserEvents.js]: yeah with like 3 if statements?

File: src/pages/member/UserEvents.js (hunk 3 above)
Comment (ianmah) [reply src/pages/member/UserEvents.js:135]: oh, gotcha

File: src/pages/member/UserEvents.js [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+
+  const { events = [], eventsRegistered = [], user } = props
+
+  useEffect(() => {
+    if (!props.events) updateEvents()
+    if (user.id) updateRegisteredEvents(user.id)
+  }, [])
+
+  const handleFavoriteEvent = async (eventId, toggle) => {
+    const body = { eventID: eventId, isFavourite: toggle }
```
Comment (ianmah) [inline src/pages/member/UserEvents.js:124]: lets keep `Favorite` vs `Favourite` consistent

**Pull request merged**